### PR TITLE
[Feature](Nereids)add some scalar functions

### PR DESCRIPTION
**Pull request by sohardforaname**
# Proposed changes

Issue Number: close #xxx

## Problem summary

add functions: running_difference, regexp_extract_all, split_by_string

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.68 seconds
 load time: 646 seconds
 storage size: 17122771036 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230104142147_clickbench_pr_73608.html

File: fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSignature.java [hunk 1]
```diff
@@ -113,4 +118,34 @@ public FunctionSignature varArgs(AbstractDataType...argTypes) {
             return FunctionSignature.of(returnType, true, argTypes);
         }
     }
+
+    public static class MultiFuncSigBuilder {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/FunctionSignature.java:122]: static number initialized when class loading. It has mulit-thread problem. So i don't think use this singleton object to build multi signature is a good idea.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/RunningDifference.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            .ret(IntegerType.INSTANCE).args(SmallIntType.INSTANCE)
+            .ret(BigIntType.INSTANCE).args(IntegerType.INSTANCE)
+            .ret(LargeIntType.INSTANCE).args(BigIntType.INSTANCE)
+            .ret(LargeIntType.INSTANCE).args(LargeIntType.INSTANCE)
+            .ret(DoubleType.INSTANCE).args(FloatType.INSTANCE)
+            .ret(DoubleType.INSTANCE).args(DoubleType.INSTANCE)
+            .ret(DecimalV2Type.SYSTEM_DEFAULT).args(DecimalV2Type.SYSTEM_DEFAULT)
+            .ret(DecimalV3Type.DEFAULT_DECIMAL32).args(DecimalV3Type.DEFAULT_DECIMAL32)
+            .ret(DecimalV3Type.DEFAULT_DECIMAL64).args(DecimalV3Type.DEFAULT_DECIMAL64)
+            .ret(DecimalV3Type.DEFAULT_DECIMAL128).args(DecimalV3Type.DEFAULT_DECIMAL128)
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/RunningDifference.java:60]: remove decimal v3 type

File: regression-test/suites/nereids_syntax_p0/function.groovy [hunk 3]
```diff
@@ -89,5 +89,59 @@ suite("function") {
...
+    """
+
+    sql """                                       
+        INSERT into running_difference_test values 
+            ('1', '2022-10-28', '2022-03-12 10:41:00', null),
+            ('2','2022-10-27', '2022-03-12 10:41:02', 2.6),
+            ('3','2022-10-28', '2022-03-12 10:41:03', 2.5),
+            ('4','2022-9-29', '2022-03-12 10:41:03', null),
+            ('5','2022-10-31', '2022-03-12 10:42:01', 3.3),
+            ('6', '2022-11-08', '2022-03-12 11:05:04', 4.7);
```
Comment (morrySnow) [inline regression-test/suites/nereids_syntax_p0/function.groovy:119]: maybe u should test all type running_differece supported.

**Comment by sohardforaname** [comment]
duplicate with https://github.com/apache/doris/pull/15671

**Pull request closed without merging**